src/js/components/CollapsibleErrorMessage.js

```diff
@@ -0,0 +1,231 @@
+    let {message, details} = this.props;
+    let isDetailed = message && details && details.length;
+
+    // Check if we must not show the detail link at all
+    if (!isDetailed) {
+      return null;
+    }
+
+    // Get text
+    let moreLess = 'more';
+    if (this.state.expanded) {
+      moreLess = 'less';
+    }
+
+    // Render
+    return (
+        <span
+          className="collapsible-toggle-label  clickable"
+          onClick={this.toggleExpanded} >
+          (Show {moreLess})
+        </span>
+      );
+
+  }
+
+  /**
+   * [Custom component] The list of errors in the details
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getDetailMessages() {
+    let {details} = this.props;
+    return details.map(function (message) {
+      let msg = message.toString();
+      return (
+          <li key={msg}>{msg}</li>
+        );
+    });
+  }
+
+  /**
+   * [Custom component] The fixed message part
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getFixedPart() {
+    let {message} = this.props;
+    let isVisible = !!message;
+
+    // If not visible, just exit
+    if (!isVisible) {
+      return null;
+    }
+
+    // Render the fixed part of the message
+    return (
+        <div className="collapsible-fixed">
+          <Icon
+            family="mini"
+            id="yield"
+            size="mini"
+            className="icon-alert icon-margin-right"
+            color="red" />
+          {message}
+          {this.getDetailLink()}
+        </div>
+      );
+  }
+
+  /**
+   * [Custom component] The toggled message part
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getToggledPart() {
+    let {message, details} = this.props;
+    let isDetailed = !!message && details && details.length;
+
+    // If not expanded or detailed, just exit
+    if (!isDetailed || !this.state.expanded) {
+      return null;
+    }
+
+    // Render the toggled part of the message
+    // (Note: The nested div is used to float the contents when needed)
+    return (
+        <div className="collapsible-toggled">
+          <div>
+            <ul>
+            {this.getDetailMessages()}
+            </ul>
+          </div>
+        </div>
+      );
+
+  }
+
+  /**
+   * React.js Render Function
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  render() {
+    let {message} = this.props;
+    let isVisible = !!message;
+
+    // If not visible, just exit
+    if (!isVisible) {
+      return null;
+    }
+
+    // Compile classes
+    let className = classNames(
+      this.props.className,
+      'collapsible-error-message',
+      {
+        'expanded': this.state.expanded
+      }
+    );
+
+    // Render message component
+    return (
+        <div className={className}>
+          {this.getFixedPart()}
+          {this.getToggledPart()}
+        </div>
+      );
+
+  };
+
+};
+
+CollapsibleErrorMessage.defaultProps = {
+  className: '',
```

Comment:
You are right, but since it's a property I wanted to have a default value. `className : undefined` might also be an acceptable option I guess, but it looks a bit uglier.

Out of curiosity, is `undefined` or `''` causing bigger (the tiniest, neglectable, yet there) performance impact?